File: Timus/1114/main.py
dp = [[[0 for x in range(22)] for y in range(22)] for z in range(22)]

def solve(c, A, B):

	if c == 0:
		return 1

	
	if dp[c][A][B] != -1:
		return dp[c][A][B]

	res = 0

	i = 0
	j = 0

	while A - i >= 0:
		j = 0
		while B - j >= 0:
			c1 = c - 1
			a1 = A - i
			b1 = B - j
			res = res + solve(c1, a1, b1)
			j = j + 1
		i = i + 1

	
	dp[c][A][B] = res;

	return res


line = input()
args = line.split()
N = int(args[0])
Aa = int(args[1])
Bb = int(args[2])

for i in range(0, N + 1):
	for j in range(0, Aa + 1):
		for k in range(0, Bb + 1):
			dp[i][j][k] = -1

print(solve(N, Aa, Bb))

File: Timus/1114/main.cpp
#include "bits/stdc++.h"
using namespace std;

#define db(x) //cerr << #x << " == " << x << endl
#define all(container) container.begin(), container.end()
#define mp(i,j) make_pair(i,j)
#define pb push_back

typedef pair<int,int> pii;
typedef unsigned long long ll;
typedef vector<ll> vll;
typedef vector<int> vi;
typedef vector<pii> vii;

template<typename T>
void print_vector_debug(const T& t) {
	cout << "[DEBUG] VECTOR:";
	for (auto i = t.cbegin(); i != t.cend(); ++i) {
		if ((i + 1) != t.cend()) {
			cout << *i << " ";
		} else {
			cout << *i << endl;
		}
	}
}

vector<vector<vll>> dp;

ll solve(int c, int A, int B)
{

	if (c == 0)
	{
		return 1;
	}
	
	
	ll &res = dp[c][A][B];
	if (res != -1)
	{
		return res;
	}
	
	res = 0;
	
	for (int i = 0; A - i >= 0; i++)
	{
		for (int j = 0; B - j >= 0; j++)
		{
			res += solve(c - 1, A - i, B - j);
		}
	}
	
	return res;
}

int main() 
{
	
	int N, A, B;

	scanf("%d%d%d", &N, &A, &B);
	
	dp.resize(N + 1, vector<vll>(A + 1, vll(B + 1, -1)));
	printf("%llu", solve(N, A, B));
	return 0;
}
